supreme/src/commonMain/kotlin/at/asitplus/signum/supreme/SignatureResult.kt

```diff
@@ -0,0 +1,41 @@
+package at.asitplus.signum.supreme
+
+import at.asitplus.KmmResult
+import at.asitplus.catching
+import at.asitplus.signum.indispensable.CryptoSignature
+import kotlin.jvm.JvmInline
+
+/** These map to SignatureResult.Failure instead of SignatureResult.Error */
+sealed class UserInitiatedCancellationReason(message: String?, cause: Throwable?): Throwable(message, cause)
+class UnlockFailed(message: String? = null, cause: Throwable? = null) : UserInitiatedCancellationReason(message, cause)
+
+sealed interface SignatureResult {
+    @JvmInline
+    value class Success(val signature: CryptoSignature): SignatureResult
+    @JvmInline
+    value class Failure(val problem: UserInitiatedCancellationReason): SignatureResult
+    @JvmInline
+    value class Error(val exception: Throwable): SignatureResult
```

Comment:
Add KDoc describing semantics of `Error` and `Failure` to avoid misinterpretation

Reply:
also;: generics to further narrow down the type of signature would be nice (e.g. to do some in-between processing and make sure every signature is rawByteEncodable, etc)

Reply:
then i lose the fact its ternary. it just needs to <T: CryptoSignature> for the success case

Reply:
but at that point you also need `transform`, `map`, `mapCatching`, ...

Reply:
see `.modify`